cg/apps/housekeeper/hk.py

```diff
@@ -399,7 +401,28 @@ def get_files_from_latest_version(self, bundle_name: str, tags: list[str]) -> Qu
         if not version:
             LOG.warning(f"Bundle: {bundle_name} not found in Housekeeper")
             raise HousekeeperBundleVersionMissingError
-        return self.files(version=version.id, tags=tags)
+        return self.files(version=version.id, tags=tags).all()
+
+    def get_files_from_latest_version_by_list_of_tags(
+        self, bundle_name: str, tags: list[set[str]], excluded_tags: list[set[str]] | None = None
+    ) -> list[File]:
+        """
+        Return files from the latest version of a bundle matching provided tags. Files containing
+        any tag sets specified in the excluded_tags list will be excluded from the output.
+        """
+        filtered_files: list[File] = []
+        if not tags:
+            return filtered_files
+        files: list[File] = self.get_files_from_latest_version(bundle_name=bundle_name)
+        for file in files:
+            file_tags: set[str] = {tag.name for tag in file.tags}
+            if any(tag.issubset(file_tags) for tag in tags):
+                if excluded_tags and any(
+                    exclude_tag.issubset(file_tags) for exclude_tag in excluded_tags
+                ):
```

Comment:
I'll rethink this function. It might need a new name and some restructuring. The other queries aim to match the provided tag with the HK tag. But here I'm allowing it to be a subset, which could confuse someone trying to reuse it 🤔